### PR TITLE
Feat/e2e test cases

### PR DESCRIPTION
**Pull request by AhmadHassanButt**
#### **Summary**
Fixed and added test cases for the `auth`, `user`, and `article` modules, including follow/unfollow functionality.

---

#### **Changes**
- **Fixed Existing Test Cases**: Resolved issues in current tests for `auth` and `user` modules.
- **Added Follow/Unfollow Tests**:
  - **Follow**: Test for successfully following a user and handling double follow attempts.
  - **Unfollow**: Test for successfully unfollowing and handling unfollowing a user not being followed.
- **Added Article Module Tests**: Implemented test cases for the `article` module functionality.

---

### CODE REVIEW

File: test/article/articles.e2e-spec.ts [hunk 1]
```diff
@@ -27,9 +33,104 @@ describe('ArticlesController (e2e)', () => {
 
   afterAll(async () => {});
 
+  it('should create a new article with random data', async () => {
+    const newArticle = {
+      body: faker.lorem.paragraph(), // Required
```
Comment (dev-ahmadbilal) [inline test/article/articles.e2e-spec.ts]: Instead of using faker directly please create a wrapper class.

File: test/admin/users.e2e-spec.ts [hunk 2]
```diff
@@ -1,8 +1,7 @@
 import request from 'supertest';
 
-import { RoleEnum } from '@src/roles/roles.enum';
-import { StatusEnum } from '@src/statuses/statuses.enum';
```
Comment (dev-ahmadbilal) [inline test/admin/users.e2e-spec.ts:4]: Absolute path's config can be updated to support in test folder as well.

File: test/user/follow-unfollow.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        email: userEmail,
+        password: userPassword,
+        firstName,
+        lastName,
+        username,
+      })
+      .expect(204);
+
+    // Create and authenticate another user to follow
+    const anotherUserEmail = `anotherUser.${Date.now()}@example.com`;
```
Comment (dev-ahmadbilal) [inline test/user/follow-unfollow.e2e-spec.ts]: Why don't we use faker here? any reason?

Comment (AhmadHassanButt) [reply test/user/follow-unfollow.e2e-spec.ts]: I added this specific file and the faker is implemented by @omerasifkhan. Let me look into it

File: test/admin/users.e2e-spec.ts (hunk 2 above)
Comment (AhmadHassanButt) [reply test/admin/users.e2e-spec.ts:4]: Let me look into it

File: test/article/articles.e2e-spec.ts (hunk 1 above)
Comment (AhmadHassanButt) [reply test/article/articles.e2e-spec.ts]: Resolved

File: test/admin/users.e2e-spec.ts (hunk 2 above)
Comment (AhmadHassanButt) [reply test/admin/users.e2e-spec.ts:4]: Resolved

File: test/user/follow-unfollow.e2e-spec.ts (hunk 3 above)
Comment (AhmadHassanButt) [reply test/user/follow-unfollow.e2e-spec.ts]: Resolved

**Review by dev-ahmadbilal — APPROVED** [review]

**Pull request merged**